### PR TITLE
Adapt prepdisk for update

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/JrA9u4mp/531-2-adapt-inst-prepdisk-for-upgrade

### CODE REVIEW

File: test/clients/inst_prepdisk_test.rb [hunk 1]
```diff
@@ -39,17 +39,36 @@
...
       allow(storage).to receive(:prepend_rootprefix).with("/sys").and_return "/dest/sys"
       allow(Yast::SCR).to receive(:Execute).and_return(true)
+      allow(Yast::Mode).to receive(:update).and_return(mode == :update)
     end
 
-    it "uses the destination directory to mount and prepare the result" do
-      expect(storage).to receive(:rootprefix=).with("/dest")
-      client.run
+    context "in normal mode" do
+      let(:mode) { :normal }
```
Comment (ancorgs) [inline test/clients/inst_prepdisk_test.rb]: Not sure if checking for `normal` mode makes much sense for this client. I would have expected `installation` and `update` to be tested, but `normal`...

File: test/clients/inst_prepdisk_test.rb [hunk 2]
```diff
@@ -39,17 +39,36 @@
...
 
-    it "commits all libstorage pending changes" do
-      expect(storage).to receive(:calculate_actiongraph)
-      expect(storage).to receive(:commit)
-      client.run
+    context "in update mode" do
+      let(:mode) { :update }
+
+      it "does not commit anything" do
+        expect(client).not_to receive(:commit)
```
Comment (ancorgs) [inline test/clients/inst_prepdisk_test.rb]: `InstPrepdisk#commit` is a protected method. I don't see a good reason to mention it in the test. Checking for `StorageManager#commit` should be enough.

File: test/clients/inst_prepdisk_test.rb [hunk 3]
```diff
@@ -39,17 +39,36 @@
...
+      allow(Yast::Mode).to receive(:update).and_return(mode == :update)
     end
 
-    it "uses the destination directory to mount and prepare the result" do
-      expect(storage).to receive(:rootprefix=).with("/dest")
-      client.run
+    context "in normal mode" do
+      let(:mode) { :normal }
+
+      it "uses the destination directory to mount and prepare the result" do
```
Comment (ancorgs) [inline test/clients/inst_prepdisk_test.rb]: I miss a "does not changes libstorage root prefix" as counterpart for this in the update mode.

File: test/clients/inst_prepdisk_test.rb (hunk 1 above)
Comment (joseivanlopez) [inline test/clients/inst_prepdisk_test.rb]: Ok, I supposed that `normal` means `installation`. Imo just explained me the differences 👍

File: test/clients/inst_prepdisk_test.rb (hunk 3 above)
Comment (joseivanlopez) [inline test/clients/inst_prepdisk_test.rb]: Ok.

File: test/clients/inst_prepdisk_test.rb (hunk 2 above)
Comment (joseivanlopez) [inline test/clients/inst_prepdisk_test.rb]: Yes, you are right. We can test it only looking at public methods.

**Comment by ancorgs** [comment]
It would be nice to also have an entry in the changelog. Version bump not needed

**Pull request merged**